src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -57,60 +57,87 @@ internal partial class SymbolTreeInfo : IObjectWritable
             Func<ObjectReader, T> tryReadObject,
             CancellationToken cancellationToken) where T : class, IObjectWritable, IChecksummedObject
         {
-            if (checksum == null) 
+            Func<Task<T>> wrappedTask = async () =>
```

Comment:
❔  How often is this called? This change introduces a capture allocation on the fast path for this method.

Reply:
Resolved by using a local function. Further analysis seems low priority considering there are several other allocations involved with this metho.d

Reply:
changed to local function